tutorials/20_REST_API.ipynb

```diff
@@ -0,0 +1,538 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Tutorial: Using Haystack with REST API\n",
+    "\n",
+    "- **Level**: Advanced\n",
+    "- **Time to complete**: 30 minutes\n",
+    "- **Prerequisites**: Basic understanding of Docker, basic knowledge of Haystack pipelines \n",
+    "- **Nodes Used**: `ElasticsearchDocumentStore`, `EmbeddingRetriever`\n",
+    "- **Goal**: Learn how you can interact with Haystack through REST API."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Overview\n",
+    "\n",
+    "Haystack enables you to apply the latest NLP technology to your own data and create production-ready applications. Building an end-to-end NLP application requires the combination of multiple concepts:\n",
+    "* **DocumentStore** stores the data. You will use Elasticsearch for this tutorial.\n",
+    "* **Haystack** pipelines convert files into [Documents](https://docs.haystack.deepset.ai/docs/documents_answers_labels#document), index them to DocumentStore, and run NLP tasks such as question answering and document search.\n",
+    "* **REST API** allows other applications to interact with Haystack by handling their queries and returning responses.\n",
+    "* **Docker** simplifies the environment setup needed to have Elasticsearch running.\n",
+    "\n",
+    "This tutorial introduces you to all the concepts needed to build an end-to-end document search application. After completing this tutorial, you will have learned how to create a pipeline YAML file, index files, and how to query your application using REST API."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Preparing the Environment"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. Update or install Docker and Docker Compose, and launch Docker."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "apt-get update && apt-get install docker && apt-get install docker-compose\n",
```

Comment:
I understand this tutorial is not meant to be run on Colab - in this case, I would not provide `bash` cells. For example, this one would fail on the average Mac setup, while another one below would fail on Linux. We should probably just point users towards the Docker install page. Something we could do, is adding a cell users can run to confirm Docker is up and running, like `docker info`